### PR TITLE
Make images for theme colour and layout clickable

### PR DESCRIPTION
**Pull request by ssha0054**
# Description 
On the Theme selector page, the images are currently not clickable. In order to select a theme colour or layout, you need to select the checkbox. This PR makes these images clickable so you now have another way of selecting a theme and layout.

# Type of change 
- [ ] Bug fix
- [x] New feature
- [x] Refactoring
- [ ] Documentation 

# Problem
Images in the theme selector were not clickable.

# Solution description
Made the images for both theme and layout have onClick actions which result in the corresponding radio button to be checked whilst also ensuring the correct information regarding theme and layout is stored in the state.

# Tests 
All unit tests pass? Yes

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
Before:
image

After clicking on the image for layout 3 and the image for the black and white theme:
image

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low

### CODE REVIEW

File: client/src/components/dashboard/TemplateSelector.js
Comment (pmyip3) [reply client/src/components/dashboard/TemplateSelector.js:100]: Not quite sure why name and values do not work but figure out that the id props work.
image

File: client/src/components/dashboard/TemplateSelector.js
Comment (pmyip3) [inline client/src/components/dashboard/TemplateSelector.js:116]: Just a suggestion for improvement is it possible to combine the `<Form.Check> `and `<div>` into a single div so we only handle the onClick function in a single place instead of 2?

**Review by pmyip3 — COMMENTED** [review]
Another suggestion: might have it in a separate ticket though as I am not quite sure if the numbers of color/ layout would be increases or not. If that is the case, this files might become large, so would be nice if we can have another file to store the relevant layout and color data like what we have in SidebarData.js

File: client/src/components/dashboard/TemplateSelector.js
Comment (jfan0010) [reply client/src/components/dashboard/TemplateSelector.js:116]: @ssha0054  Good news, I found a way to wrap them, see [here](https://stackoverflow.com/questions/19412204/how-do-you-make-the-radio-button-text-to-be-clickable-too/54573527)

File: client/src/components/dashboard/TemplateSelector.js
Comment (jfan0010) [reply client/src/components/dashboard/TemplateSelector.js:100]: @ssha0054 just found out that using `label` you can wrap them together, and doesn't need to implement two `onClick` or `onChange` in one component. See: https://stackoverflow.com/questions/19412204/how-do-you-make-the-radio-button-text-to-be-clickable-too/54573527

**Review by ssha0054 — COMMENTED** [review]
**Changes made:** Used a `label` to wrap the radio buttons and images so now only one `onChange`/`onClick` call is needed. Also removed hardcoding in the values when calling the `updateSelections` function, however had some issue with the `value` being used in the case statements so decided to use the `id` instead. These changes have also made the theme image to now be below the radio button instead of being on the side. Refer to pic below: 
image
Not sure how to fix this atm but will make sure it is back to how it was originally before merging.

File: client/src/components/dashboard/TemplateSelector.js
Comment (pmyip3) [inline client/src/components/dashboard/TemplateSelector.js:30]: empty lines in line 30 and 32

**Review by pmyip3 — APPROVED** [review]

**Review by aksh0001 — APPROVED** [review]
Will defer to @jfan0010 for the React practices.

**Comment by aksh0001** [comment]
Actually, I wanted to add that this entire UI may need to be refactored as we're thinking of implementing a toggle between light/dark theme as opposed to color selection.
@vgoh1 @sanathmahesh @rsal0007 @melvinalex7 @melvinalex77

**Comment by ssha0054** [comment]
> Actually, I wanted to add that this entire UI may need to be refactored as we're thinking of implementing a toggle between light/dark theme as opposed to color selection.
> @vgoh1 @sanathmahesh @rsal0007 @melvinalex7 @melvinalex77

In that case, should this PR still get merged into main as the theme layout images being clickable would still be useful for later (even with any changes to this UI)? Or would it be better to just close this PR and let whoever works on the implementation of the toggle to also implement the ability to click on any images, if required.

**Comment by vgoh1** [comment]
@ssha0054 I think it's worth merging because the redesign is still far away in the future. The theme layout pictures should still be useful.

**Review by vgoh1 — APPROVED** [review]
Helped fix a bug (L52 in TemplateSelector.js) that caused the site to crash when you hit the create website button in the theme selector. :) LGTM otherwise.

**Pull request merged**
